### ISSUE TITLE
Time tracking and reporting based on labels

### ISSUE THREAD

**Issue by tj** [ZenHubIO/support#596]
For me the use-case is not so much for billing (#594), but to see where time really goes. Either by tragging in/out of a column or timer button on individual issues, track the time, and later report based on the labels of each issue.

I think this would give you a really good look at where time is spent without using external tools which become redundant if you need to create all the same labels etc.

**Comment by tj** [comment]
This could also over time provide you with time estimates based on points, since it would learn how accurate you are with scoring.

**Comment by azenMatt** [comment]
Thanks @tj - we are currently looking into what new reports would be useful to introduce into ZenHub, we will take a look into this idea. 

I think we could do some interesting cycle time reports based on a combination of pipeline status and labels :+1:

**Comment by tj** [comment]
FWIW I'm happy to implement some of this stuff if ZenHub can spare a few bucks :D haha.

**Comment by azenMatt** [comment]
That would be amazing @tj 😄  - we will get in touch 

Improving estimation is one of our big Q1 goals. In particular how we can better show estimate vs actual and use that to help teams become better at estimating.

**Comment by enfluid** [comment]
We are definitely interested in some kind of time tracking.  Really, for us, it's about hours.  We want to estimate a task in hours, and then track accurately how many hours that task takes, and then use that hourly data to pay developers and bill clients.  Is something like that in the pipeline?  Currently, we're planning on using EverHour for that, but we'd love to have it all in a single system.

**Comment by azenMatt** [comment]
Hi @enfluid - that is one of the biggest blockers for teams using ZenHub right now. It does seem clear that the current 'agile story points' only methodology isn't the right fit for all teams. 

we need to do a lot more discovery work, for which we will keep this repo updated, but ideally we would like to be able to improve 

1) how teams are able to estimate (include hours/days as well as story points)
2) how teams are able to track where the time went 
3) how teams are able to track what they estimated vs what it actually took

**Comment by enfluid** [comment]
Awesome, thanks @azenMatt!  We eagerly await additional news.  Let us know if you'd like our input on what would be helpful!  You can contact me via:  http://www.enfluid.com/

**Comment by shacharz** [comment]
Hi, 

We'd also see big value in a feature that'll allow you to measure time estimated vs time spent.
1. I think initially, nothing automated is needed, I just want to be able to report how much time I've just spent on a specific task, just like the estimate feature (only cumulative). So I can see per ticket, time estimated time spent.
2. The next steps would be to have it on the Milestone burndown report, and see how much time was spent vs how much story points go down (so you can see how good or badly your sprint is shaping)
3. automatic time calculation closely depends on your workflow, so that would probably require more research into how different teams use zenhub, and I'd leave it for last.

Note: Some teams might like to see how much time the PR that took closed an issue took, so that's just another way of looking at things.

Tl;DR: duplicate estimate story points, to 'story points spent'

**Comment by tj** [comment]
Ditto, I wouldn't mind even just having a start/stop timer thing and reporting against that and labels.

**Comment by shacharz** [comment]
I wouldn't mind even no timers at all, just me adding up story points that I've spent.
